Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs

```diff
@@ -0,0 +1,222 @@
+        }
+    }
+
+    /// <summary>
+    /// Contains a number of methods to assert that a <see cref="HttpResponseMessage"/> is in the expected state.
+    /// </summary>
+    [DebuggerNonUserCode]
+    public class HttpResponseMessageAssertions<TAssertions>
+        where TAssertions : HttpResponseMessageAssertions<TAssertions>
+    {
+        public HttpResponseMessageAssertions(HttpResponseMessage value) => Subject = value;
+
+        /// <summary>
+        /// Gets the object which value is being asserted.
+        /// </summary>
+        public HttpResponseMessage Subject { get; }
+
+        /// <summary>
+        /// Asserts that the <see cref="HttpStatusCode"/> is successful (2xx).
+        /// </summary>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> BeSuccessful(string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .ForCondition(Subject is not null)
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected HttpStatusCode to be successful (2xx){reason}, but HttpResponseMessage was <null>.");
+
+            Execute.Assertion
+                .ForCondition(Subject.IsSuccessStatusCode)
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected HttpStatusCode to be successful (2xx){reason}, but found {0}.", Subject.StatusCode);
+
+            return new AndConstraint<TAssertions>((TAssertions)this);
+        }
+
+        /// <summary>
+        /// Asserts that the <see cref="HttpStatusCode"/> is redirection (3xx).
+        /// </summary>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> BeRedirection(string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .ForCondition(Subject is not null)
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected HttpStatusCode to be redirection (3xx){reason}, but HttpResponseMessage was <null>.");
+
+            Execute.Assertion
+                .ForCondition((int)Subject.StatusCode >= 300 && (int)Subject.StatusCode <= 399)
```

Comment:
Suggestion
```cs
.ForCondition((int)Subject.StatusCode is >= 300 and <= 399)
```

Reply:
I love C# 9.0 ;-)

Reply:
Yes me too! They just have to be integrated into my muscle memory ;)